wolfcrypt/src/wc_pkcs11.c

```diff
@@ -559,12 +571,61 @@ int wc_Pkcs11Token_Init(Pkcs11Token* token, Pkcs11Dev* dev, int slotId,
         token->userPinSz = (CK_ULONG)userPinSz;
     }
 
-    if (slot != NULL)
+    if (slot != NULL) {
         XFREE(slot, dev->heap, DYNAMIC_TYPE_TMP_BUFFER);
+    }
 
     return ret;
 }
 
+/**
+ * Set up a token for use.
+ *
+ * @param  [in]  token      Token object.
+ * @param  [in]  dev        PKCS#11 device object.
+ * @param  [in]  slotId     Slot number of the token.<br>
+ *                          Passing -1 uses the first available slot.
+ * @param  [in]  tokenName  Name of token to initialize (optional)
+ * @param  [in]  userPin    PIN to use to login as user.
+ * @param  [in]  userPinSz  Number of bytes in PIN.
+ * @return  BAD_FUNC_ARG when token, dev and/or tokenName is NULL.
+ * @return  WC_INIT_E when initializing token fails.
+ * @return  WC_HW_E when another PKCS#11 library call fails.
+ * @return  0 on success.
+ */
+int wc_Pkcs11Token_Init(Pkcs11Token* token, Pkcs11Dev* dev, int slotId,
+    const char* tokenName, const unsigned char* userPin, int userPinSz)
+{
+    int tokenSz = 0;
+    if (tokenName != NULL) {
+        tokenSz = (int)XSTRLEN(tokenName);
```

Comment:
You could make tokenSz a size_t and then the conversion would be unnecessary. It also makes more sense for the size to be unsigned, IMO.

Reply:
Done. Thanks for the suggestion!